libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/snapshotter/api/MapboxSnapshotterApi.kt

```diff
@@ -120,10 +121,24 @@ class MapboxSnapshotterApi(
                         snapshotter.setCameraOptions(cameraOptions)
                         snapshotter.setUri(options.styleUri)
                         mapInterface.size = oldSize
-                        snapshotter.start(object : Snapshotter.SnapshotReadyCallback {
-                            override fun onSnapshotCreated(
-                                snapshot: Expected<MapSnapshotInterface?, String?>
-                            ) {
+                        snapshotter.setStyleListener(object : SnapshotStyleListener {
+                            override fun onDidFinishLoadingStyle(style: Style) {
+                                style.addSource(
+                                    geoJsonSource(RouteConstants.PRIMARY_ROUTE_SOURCE_ID) {
+                                        geometry(LineString.fromLngLats(routeLinePoints))
+                                    }
+                                )
+                                style.addLayer(
+                                    (
+                                        SnapshotterProcessor
+                                            .process(SnapshotterAction.GenerateLineLayer)
+                                            as SnapshotterResult.SnapshotLineLayer
+                                        ).layer
+                                )
+                            }
+                        })
+                        snapshotter.start(object : SnapshotCreatedListener {
+                            override fun onSnapshotResult(snapshot: MapSnapshotInterface?) {
```

Comment:
cc @abhishek1508, I wasn't able to test because I don't have the correct token. Let's sync about this.

Reply:
Verified that this still works:
![device-2021-03-05-185905](https://user-images.githubusercontent.com/16925074/110154784-f1f03980-7de4-11eb-920b-32272cf2bd4a.png)